routers/web/repo/issue.go

```diff
@@ -2768,6 +2768,36 @@ func NewComment(ctx *context.Context) {
 			if form.Status == "reopen" && issue.IsPull {
 				pull := issue.PullRequest
 				var err error
+				// Check whether the head commit of PR exists in base branch
+				// Get head commit ID of PR
+				prHeadRef := pull.GetGitRefName()
+				if err := pull.LoadBaseRepo(ctx); err != nil {
+					ctx.ServerError("Unable to load base repo", err)
+					return
+				}
+				prHeadCommitID, err := git.GetFullCommitID(ctx, pull.BaseRepo.RepoPath(), prHeadRef)
+				if err != nil {
+					ctx.ServerError("Get head commit Id fail", err)
```

Comment:
Should we return 500 directly? Maybe just returning a flash with error is better.